Rust/src/common/anomalydescriptor.rs

```diff
@@ -0,0 +1,120 @@
+use std::cmp::min;
+use crate::common::divector::DiVector;
+
+/**
+ * This class maintains a simple discounted statistics. Setters are avoided
+ * except for discount rate which is useful as initialization from raw scores
+ */
+#[repr(C)]
+#[derive(Clone)]
+pub struct AnomalyDescriptor {
+
+    // the current input point; can have missing values
+    pub current_input: Vec<f32>,
+
+    // current timestamp
+    pub current_timestamp: usize,
+
+    // potential missing values in the current input (ideally None)
+    pub missing_values: Option<Vec<i32>>,
+
+    // potentially transformed point used by RCF, can have different dimensions than input
+    pub rcf_point: Option<Vec<f32>>,
+
+    pub score: f32,
+
+    pub internal_timestamp: usize,
+
+    pub threshold: f32,
+
+    pub anomaly_grade: f32,
+
+    pub data_confidence: f32,
+
+    // present only if grade > 0
+    pub attribution: Option<DiVector>,
+
+    pub expected_rcf_point: Option<Vec<f32>>,
+
+    pub relative_index: Option<usize>,
+
+    // useful for time augmented forests
+    pub expected_timestamp: Option<usize>,
+
+    pub start_of_anomaly: Option<bool>,
+
+    pub in_high_score_region: Option<bool>,
+
+    pub relevant_attribution: Option<Vec<f32>>,
+
+    pub time_attribution: Option<f32>,
+
+    // the values being replaced; may correspond to past
+    pub past_values: Option<Vec<f32>>,
+
+    pub past_timestamp: Option<usize>,
+
+    pub expected_values_list: Option<Vec<Vec<f32>>>,
+
+    pub likelihood_of_values: Option<Vec<f32>>
+
+}
+
+impl AnomalyDescriptor {
+    pub fn default(point: Vec<f32>, timestamp: usize) -> Self {
```

Comment:
In Rust, `default` is typically used to define 0-argument constructors. Should we call this `new` and rename the second constructor to `new_with_missing_values` or something like that?

Reply:
done